### ISSUE TITLE
detect preview buffer display size

### ISSUE THREAD

**Issue by gertcuykens** [junegunn/fzf#1544]
Preview scroll works fine now in master after your latest commit and doesn't scroll anymore when it fits on screen.

The only remark I have left is that it would be perfect if you could calculate the preview screen size and use it as scroll offset so you can't overscroll a buffer that doesn't fit on a screen. For example
 
![image](https://user-images.githubusercontent.com/863731/55284467-9ce55680-5377-11e9-9781-5ae2c320e3f2.png)

Can be overscrolled like this

![image](https://user-images.githubusercontent.com/863731/55284471-b090bd00-5377-11e9-8ae1-337675ac11fd.png)

The perfect way would be that it stop scrolling when the last line is visible like so

![image](https://user-images.githubusercontent.com/863731/55284524-a9b67a00-5378-11e9-8fdc-dbbcc37ef2ff.png)

So basically the first preview screenshot should be offset by the number of lines visible so in this case 28 / 43 instead of 1 / 43. This will result in 43 / 43 by scrolling just a few lines from 28 to 43, effectively preventing overscroll of empty lines

- Category
    - [x] fzf binary
    - [ ] fzf-tmux script
    - [ ] Key bindings
    - [ ] Completion
    - [ ] Vim
    - [ ] Neovim
    - [ ] Etc.
- OS
    - [ ] Linux
    - [x] Mac OS X
    - [ ] Windows
    - [ ] Windows Subsystem for Linux
    - [ ] Etc.
- Shell
    - [x] bash
    - [ ] zsh
    - [ ] fish

**Comment by junegunn** [comment]
This is surprisingly hard to get right with `--preview-window wrap`, because of long lines that can span multiple lines. With the presence of tab characters and wide characters, it's not possible to know the actual display width of a line without going through each character.

To make matters worse, if you use mouse wheel to scroll the window, the offset can be changed by multiple lines at once, and we'll have to "gradually pull down" the window if we realize we accidentally over-scrolled and the content didn't reach the bottom. We then have to consider the actual number of lines each line above the screen will occupy. Think about a document with long lines here and there.

So I've decided that it's not really worth the effort, especially since the over-scroll issue doesn't bother me that much. I'd like to keep the code simple rather than trying too hard to perfect it.

**Comment by antoinemadec** [comment]
@junegunn , detecting the preview buffer display size would also be great to have an option to **center** the preview window:
```
--preview-window=center:+{2}
```

I followed your **fzf.vim** in [coc-fzf](https://github.com/antoinemadec/coc-fzf). I can know scroll the preview window which is great.
However, the ```+{2}-5``` always gives me an offset of 5, no matter the size of the preview window.

**Would this center feature could be added in Fzf?**
